### PR TITLE
#1072 registration: backend form validations do not crash anymore

### PR DESCRIPTION
**Pull request by mouneyrac**

### CODE REVIEW

**Comment by mouneyrac** [comment]
As I am not expert in Django, I would understand there may be a better solution.

Basically I switched the "return self.form_invalid" to "return render", which is what we do in other places in RDRF.
I have not seen this "return self.form_invalid" (on the base class as in this code) neither in our code or in the documentation - we do have one "return self.form_invalid" in our code but it is intercepted by our own code and transform in render. So I changed for "return render".

Once I did it I found out the problem came from the missing information about the registry_code.

**Comment by mouneyrac** [comment]
Tested on Angelman.

File: rdrf/rdrf/views/registration_rdrf.py [hunk 1]
```diff
@@ -45,7 +45,9 @@ def post(self, request, *args, **kwargs):
         if form.is_valid():
             return self.form_valid(form)
         else:
-            return self.form_invalid(form)
+            context = self.get_context_data(form=form)
```
Comment (id2359) [inline rdrf/rdrf/views/registration_rdrf.py]: Add logging form validation errors here ( for error in form.errors )

Comment (mouneyrac) [reply rdrf/rdrf/views/registration_rdrf.py]: done

Comment (mouneyrac) [reply rdrf/rdrf/views/registration_rdrf.py]: example of log:
runserver_1    | [WARNING:2019-11-04 15:55:46,367:registration_rdrf.py:56:post] Backend validation of the registration has failed: {'username': [{'message': 'Custom user with this Username already exists.', 'code': 'unique'}]}

**Pull request merged**